### PR TITLE
Close solicitor results

### PR DESCRIPTION
**Pull request by ansonK**
Adds ways of clearing own solicitor search results:
* pressing ESC key
* clicking a Close link in the results

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -48,3 +48,9 @@
 .solicitor_results_list {
   padding-left: 14px;
 }
+
+[data-fn="btn-solicitor-results-close"] {
```
Comment (pixeltrix) [inline app/assets/stylesheets/application.scss]: Is a data attribute selector going to work in IE7? I don't know - just asking the question.

File: app/views/defence_requests/new.html.erb [hunk 2]
```diff
@@ -82,6 +82,13 @@
 
 <script>
  $(document).ready(function () {
+
+  $(document).on('keydown', function(event) {
+    if($('.solicitor_results_list *').length > 0 && event.keyCode === 27){
```
Comment (pixeltrix) [inline app/views/defence_requests/new.html.erb]: Not sure whether it'll be a problem but sometime different keyboards report different values: http://unixpapa.com/js/key.html

I think `ESC` is consistent though so it should be okay

File: spec/features/defence_request_creation_spec.rb [hunk 3]
```diff
@@ -91,6 +91,36 @@
...
+     visit root_path
+     click_link 'New Defence Request'
+     choose 'Own'
+     fill_in 'q', with: "Bob Smith"
+
+     click_button 'Search'
+     expect(page).to have_content 'Bobson Smith'
+
+     within('.solicitor_results_list') do
+      click_link("Close")
```
Comment (pixeltrix) [inline spec/features/defence_request_creation_spec.rb]: Just nitpicking but missing a space on the indent

File: spec/features/defence_request_creation_spec.rb [hunk 4]
```diff
@@ -91,6 +91,36 @@
...
+     end
+     expect(page).not_to have_content 'Bobson Smith'
+  end
+
+  scenario "pressing ESC clears solicitor search results", js: true do
+    stub_solicitor_search_for_bob_smith
+     visit root_path
```
Comment (pixeltrix) [inline spec/features/defence_request_creation_spec.rb]: Again just nitpicking but indents messed up

File: spec/features/defence_request_creation_spec.rb (hunk 3 above)
Comment (ansonK) [inline spec/features/defence_request_creation_spec.rb]: yeah not sure whose editor is not set to 2 spaces, was something I was going to raise

File: app/assets/stylesheets/application.scss (hunk 1 above)
Comment (ansonK) [inline app/assets/stylesheets/application.scss]: according to https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors IE7 does support the [name=value] selector so should be ok

File: app/views/defence_requests/new.html.erb (hunk 2 above)
Comment (ansonK) [inline app/views/defence_requests/new.html.erb]: afaik ESC is consistent. If there is an issue with it there is still the Close link as a fallback
I'm going to look into something like browserstack for beta so we can be more certain about these sorts of things

File: app/views/defence_requests/new.html.erb [hunk 5]
```diff
@@ -1,2 +1 @@
-<%= render 'shared/form_template', form_type: 'new', object_name: 'defence_request' %>
-
+<%= render 'shared/form_template', form_type: 'new', object_name: 'defence_request' %>
```
Comment (FriedSock) [inline app/views/defence_requests/new.html.erb]: We refactored the form template so you will need to pass in `klass: DefenceRequest` after rebasing.

**Pull request merged**